src/middlewares/auth.middleware.ts

```diff
@@ -0,0 +1,37 @@
+import { Request, Response, NextFunction } from 'express';
+import * as jwt from 'jsonwebtoken';
+
+
+export const checkJwt = (req: Request, res: Response, next: NextFunction) => {
+  //Get the jwt token from the head
+  const bearerToken = <string>req.headers.authorization;
+  let jwtPayload;
+
+  //Try to validate the token and get data
+  try {
+    if (bearerToken != undefined) {
+      const token = bearerToken.split(' ')[1];
+      jwtPayload = <any>jwt.verify(token, process.env.Secret_Key_JWT!);
+      res.locals.jwtPayload = jwtPayload;
+    } else {
+      //If token is not valid or not existing, respond with 401 (unauthorized)
+      res.sendStatus(401);
+    }
+
+  } catch (error) {
+    //If token is not valid or not existing, respond with 401 (unauthorized)
+    res.sendStatus(401);
+
+  }
+
+  //The token is valid for 1 hour
+  //We want to send a new token on every request
+  const { idUser, login } = jwtPayload;
+  const newToken = jwt.sign({ idUser, login }, process.env.Secret_Key_JWT!, {
+    expiresIn: '1h'
+  });
+  res.setHeader('tokenFormatech', newToken);
```

Comment:
Maybe this one i don't know if it's a good thing. But we will see how to intercept it in the front!

Reply:
I don't think it's a problem because we only have specific AM that connect to this platform...

Reply:
Yes but i don't know if we should send it on res' body or on the header..